src/uma-ctf-adapter.ts

```diff
@@ -57,25 +59,115 @@ export function handleQuestionResolved(event: QuestionResolved): void {
   entity.save();
 }
 
-export function handleAncillaryDataUpdated(call: PostUpdateCall): void {
-  log.info("update question {}", [call.inputs.questionID.toHexString()]);
+
+function isModerator(modAddress: string) : boolean {
+  let mod = Moderator.load(modAddress);
+  if(mod == null) {
+    return false;
+  }
+  return true;
+}
+
+function handleRevisionPostUpdate(call: PostUpdateCall): void {
+  let questionId = call.inputs.questionID.toHexString();
+  log.info("handling revision postUpdate question {}", [questionId]);
+
+  let modAddress = call.transaction.from.toHexString();
+
+  // Ensure that the caller is a moderator
+  if(isModerator(modAddress)) {
+    return;
+  }
+
+  // Revision entities only get created before a market is approved
+  let mkt = MarketResolution.load(questionId);
+  if(mkt == null || mkt.approved) {
+    return;
+  } 
+
+  // Revision key: questionId + transactionIndex + update hex
+  let revision = new Revision(
+    questionId + "-" + 
+    call.transaction.index.toString() + "-"  
+    + call.inputs.update.toHexString()
+  );
+  revision.questionId = questionId;
+  revision.moderator = modAddress;
+  revision.timestamp = call.block.timestamp;
+  revision.update = call.inputs.update.toString();
+  revision.transactionHash = call.transaction.hash.toHexString();
+  revision.save();
+  return;
+}
+
+function handleApprovalPostUpdate(call: PostUpdateCall): void {
+  let questionID = call.inputs.questionID.toHexString();
+  log.info("handling approval postUpdate question {}", [questionID]);
+  let modAddress = call.from.toHexString();
+
+  if(isModerator(modAddress)) {
+    return;
+  }
+
+  let mkt = MarketResolution.load(questionID);
+  if(mkt == null) {
+    return;
+  }
+
+  mkt.approved = true;
+  mkt.save();
+  return;
+}
+
+function handlePostUpdate(call: PostUpdateCall): void {
+  let questionID = call.inputs.questionID.toHexString();
+
+  log.info("handling postUpdate question {}", [questionID]);
   let entity = MarketResolution.load(call.inputs.questionID.toHexString());
 
   if (entity == null) {
     return;
   }
 
-  if (call.from != Address.fromBytes(entity.author)) {
-    // only consider updates from question author
+  let caller = call.from;
+
+  // If approved and the caller is not a moderator return
+  // only consider updates from moderators after approval
+  if (entity.approved && !isModerator(caller.toHexString())) {
+    return;
+  }
+
+  // If not approved and the caller is not the author return
+  // only consider updates from the author before approval
+  if (!entity.approved && caller != Address.fromBytes(entity.author)) {
```

Comment:
I think we actually just never want to allow the creator to make updates. They are responsible for first draft and that's it.